my-sms/spec/models/course_spec.rb

```diff
@@ -23,4 +24,14 @@
     let(:course_params) { { start_date: 3.days.ago, end_date: 3.days.ago } }
     it { is_expected.to be_valid }
   end
+
+  context 'with an invalid institute' do
+    let(:institute) do
+      institute = build(:institute, { name: nil })
+      institute.save(validate: false)
+      institute
+    end
+    let(:course_params) { { institute_id: institute.id } }
+    it { is_expected.to be_invalid }
```

Comment:
@josh-cooper   I can see what your attempting here
It is a valid thought process.

Give that a course belongs_to an institute,
you are testing that a course with an empty institute is invalid.

I would argue that you are re-testing the belongs_to association
that has already been tested in the Active Record test in the Core Rails Rspec suite.
ie .. I dont see that this test adds any value.

You could, when the Controller functionality is added, cover this case there and confirm
that for an empty institute on a course, an error msg is raised for the user.

Reply:
I'm more trying to test that the model validations are also running on the association via `validates_associated`, which is a separate API to the belong_to association. An institute that on a course that exists but has no name should be invalidated.